summingbird-storm/src/main/scala/com/twitter/summingbird/storm/BaseBolt.scala

```diff
@@ -151,11 +151,13 @@ case class BaseBolt[I, O](jobID: JobId,
   }
 
   private def finish(inputs: Chain[InputState[Tuple]], results: TraversableOnce[O]) {
-    val tuples = inputs.iterator.map(_.state).toList
+    var numTuples = Option.empty[Int]
     var emitCount = 0
     if (hasDependants) {
       if (anchorTuples.anchor) {
         results.foreach { result =>
+          val tuples = inputs.iterator.map(_.state).toList
```

Comment:
Yeah, good idea. I believe this was how it was in original code too so I didn't notice. Let me try to fix this.